lib/msal-node/src/cache/TokenCache.ts

```diff
@@ -97,6 +97,24 @@ export class TokenCache implements ISerializableTokenCache, ITokenCache {
         return this.storage.getCache();
     }
 
+    /**
+     * API that restores cache context from persistence
+     */
+    async restoreCacheContext(): Promise<void> {
```

Comment:
This code overlaps a lot with `getAccount` and `removeAccount` now. I propose we refactor them, like:

```typescript
    private async handleCacheAccess<T>(action: () => Promise<T>, hasCacheChanged: boolean): Promise<T> {
        let cacheContext;
        try {
            if (this.persistence) {
                cacheContext = new TokenCacheContext(this, hasCacheChanged);
                await this.persistence.beforeCacheAccess(cacheContext);
            }
            return await action();
        } finally {
            if (this.persistence && cacheContext) {
                await this.persistence.afterCacheAccess(cacheContext);
            }
        }
    }
    
    async restoreCacheContext(): Promise<void> {
        this.logger.trace("refreshCacheContext called");
        return this.handleCacheAccess(async () => { }, false);
    }

    async getAllAccounts(): Promise<AccountInfo[]> {
        this.logger.trace("getAllAccounts called");
        return this.handleCacheAccess(async () => this.storage.getAllAccounts(), false);
    }

    async removeAccount(account: AccountInfo): Promise<void> {
        this.logger.trace("removeAccount called");

        return this.handleCacheAccess(async () => {
            await this.storage.removeAccount(AccountEntity.generateAccountCacheKey(account));
        }, true);
    }
```

what do you think? @sameerag @bgavrilMS

Reply:
Looks good for me. Please test the non-persistent cases for `removeAccount` carefully. Also some refactor of cache is happening now, may be it will impact you/may be not. Please watch out `msal-common` code that could change.

Reply:
@derisen @tnorling made changes on cache lookups (includes accounts and tokens). Can you please update this PR with the latest branch and I can sign off?